airbyte-integrations/connectors/source-shopify/source_shopify/transform.py

```diff
@@ -0,0 +1,106 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from decimal import Decimal
+from typing import Any, Iterable, List, Mapping, MutableMapping
+
+
+class Transformer:
+    @staticmethod
+    def _get_json_types(value_type) -> List[str]:
+        json_types = {
+            str: ["string"],
+            int: ["integer", "number"],
+            float: ["number"],
+            dict: ["object"],
+            list: ["array"],
+            bool: ["boolean"],
+            type(None): [
+                "null",
+            ],
+        }
+        return json_types.get(value_type)
+
+    @staticmethod
+    def _extract_schema_type(properties: Mapping[str, Any]) -> str:
+        schema_types = properties.get("type", [])
+        if not isinstance(schema_types, list):
+            schema_types = [
+                schema_types,
+            ]
+        return schema_types
+
+    @staticmethod
+    def _find_schema_type(schema_types: List[str]) -> str:
+        not_null_types = schema_types.copy()
+        if "null" in not_null_types:
+            not_null_types.remove("null")
+        return not_null_types[0]
+
+    @staticmethod
+    def _transform_number(value: Any):
+        return Decimal(value)
+
+    def _transform_array(self, array: List[Any], item_properties: Mapping[str, Any]):
+        # iterate over items in array, compare schema types and convert if necessary.
+        item_types = self._extract_schema_type(item_properties)
+        if item_types:
+            schema_type = self._find_schema_type(item_types)
+            nested_properties = item_properties.get("properties", {})
+            item_properties = item_properties.get("items", {})
+            for item in array:
+                if schema_type == "object":
+                    self._transform_object(item, nested_properties)
+                if schema_type == "array":
+                    self._transform_array(item, item_properties)
+
+    def _transform_object(self, transform_object: MutableMapping[str, Any], properties: Mapping[str, Any]):
```

Comment:
can we have unit tests for this? `pytest --cov` could help with covering all branches